### PR TITLE
YETUS-880. add support for prototool lint

### PR DESCRIPTION
**Pull request by aw-was-here**

### CODE REVIEW

**Comment by apache-yetus** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://builds.apache.org/job/yetus-github-multibranch/job/PR-62/1/console in case of problems.

**Comment by apache-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 899 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| 0 | mvndep | 12 | Maven dependency ordering for branch |
| +1 | mvninstall | 84 | master passed |
| +1 | javadoc | 21 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 5 | Maven dependency ordering for patch |
| +1 | mvninstall | 16 | the patch passed |
| +1 | hadolint | 3 | There were no new hadolint issues. |
| +1 | markdownlint | 2 | There were no new markdownlint issues. |
| +1 | shellcheck | 0 | There were no new shellcheck issues. |
| +1 | shelldocs | 2 | There were no new shelldocs issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | javadoc | 13 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 7 | precommit in the patch passed. |
| +1 | unit | 6 | asf-site-src in the patch passed. |
| +1 | asflicense | 14 | The patch does not generate ASF License warnings. |
| | | 1113 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/yetus-github-multibranch/job/PR-62/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/yetus/pull/62 |
| Optional Tests | dupname asflicense markdownlint javac javadoc unit hadolint shellcheck shelldocs |
| uname | Linux 39dae2af95c6 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | precommit/src/main/shell/personality/yetus.sh |
| git revision | master / b4f147a |
| Default Java | 1.8.0_212 |
|  Test Results | https://builds.apache.org/job/yetus-github-multibranch/job/PR-62/1/testReport/ |
| Max. process+thread count | 99 (vs. ulimit of 2000) |
| modules | C: precommit asf-site-src U: . |
| Console output | https://builds.apache.org/job/yetus-github-multibranch/job/PR-62/1/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.6.0 hadolint=1.16.2-0-g3b6f5a6 markdownlint=0.15.0 |
| Powered by | Apache Yetus 0.11.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Comment by apache-yetus** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://builds.apache.org/job/yetus-github-multibranch/job/PR-62/2/console in case of problems.

**Comment by apache-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 1031 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| 0 | mvndep | 11 | Maven dependency ordering for branch |
| +1 | mvninstall | 81 | master passed |
| +1 | javadoc | 20 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 4 | Maven dependency ordering for patch |
| +1 | mvninstall | 17 | the patch passed |
| +1 | hadolint | 3 | There were no new hadolint issues. |
| +1 | markdownlint | 2 | There were no new markdownlint issues. |
| +1 | shellcheck | 0 | There were no new shellcheck issues. |
| +1 | shelldocs | 3 | There were no new shelldocs issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | javadoc | 13 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 8 | precommit in the patch passed. |
| +1 | unit | 6 | asf-site-src in the patch passed. |
| +1 | asflicense | 13 | The patch does not generate ASF License warnings. |
| | | 1308 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/yetus-github-multibranch/job/PR-62/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/yetus/pull/62 |
| Optional Tests | dupname asflicense markdownlint javac javadoc unit hadolint shellcheck shelldocs |
| uname | Linux 124b047a5614 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | precommit/src/main/shell/personality/yetus.sh |
| git revision | master / e7a0625 |
| Default Java | 1.8.0_212 |
|  Test Results | https://builds.apache.org/job/yetus-github-multibranch/job/PR-62/2/testReport/ |
| Max. process+thread count | 102 (vs. ulimit of 2000) |
| modules | C: precommit asf-site-src U: . |
| Console output | https://builds.apache.org/job/yetus-github-multibranch/job/PR-62/2/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.6.0 hadolint=1.16.2-0-g3b6f5a6 markdownlint=0.15.0 |
| Powered by | Apache Yetus 0.11.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

File: precommit/src/main/shell/test-patch.d/prototool.sh [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+  big_console_header "prototool plugin: ${BUILDMODE}"
+
+  start_clock
+
+  # add our previous elapsed to our new timer
+  # by setting the clock back
+  offset_clock "${PROTOTOOL_TIMER}"
+
+  echo "Running prototool against identified protobuf files."
+    if [[ -n "${EXCLUDE_PATHS_FILE}" ]] && [[ -f "${EXCLUDE_PATHS_FILE}" ]]; then
```
Comment (busbey) [inline precommit/src/main/shell/test-patch.d/prototool.sh]: nit: indentation is wrong

File: precommit/src/main/shell/test-patch.d/prototool.sh [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+  declare statstring
+
+  if ! verify_needed_test prototool; then
+    return 0
+  fi
+
+  prototool_executor patch
+
+  # shellcheck disable=SC2016
+  PROTOTOOL_VERSION=$("${PROTOTOOL}" version 2>/dev/null | "${GREP}" Version | "${AWK}" '{print $NF}')
```
Comment (busbey) [inline precommit/src/main/shell/test-patch.d/prototool.sh]: Any particular reason we do this here instead of in precheck?

Comment (aw-was-here) [reply precommit/src/main/shell/test-patch.d/prototool.sh]: Not really.  I'll move it.

**Comment by apache-yetus** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://builds.apache.org/job/yetus-github-multibranch/job/PR-62/3/console in case of problems.

**Comment by apache-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 331 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ master Compile Tests _ |
| 0 | mvndep | 11 | Maven dependency ordering for branch |
| +1 | mvninstall | 85 | master passed |
| +1 | javadoc | 21 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 4 | Maven dependency ordering for patch |
| +1 | mvninstall | 17 | the patch passed |
| +1 | hadolint | 3 | There were no new hadolint issues. |
| +1 | markdownlint | 2 | There were no new markdownlint issues. |
| +1 | shellcheck | 0 | There were no new shellcheck issues. |
| +1 | shelldocs | 3 | There were no new shelldocs issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | javadoc | 14 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 8 | precommit in the patch passed. |
| +1 | unit | 6 | asf-site-src in the patch passed. |
| +1 | asflicense | 13 | The patch does not generate ASF License warnings. |
| | | 546 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/yetus-github-multibranch/job/PR-62/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/yetus/pull/62 |
| Optional Tests | dupname asflicense markdownlint javac javadoc unit hadolint shellcheck shelldocs |
| uname | Linux 6ba39df7ab1f 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | precommit/src/main/shell/personality/yetus.sh |
| git revision | master / 90c0cd0 |
| Default Java | 1.8.0_212 |
|  Test Results | https://builds.apache.org/job/yetus-github-multibranch/job/PR-62/3/testReport/ |
| Max. process+thread count | 99 (vs. ulimit of 2000) |
| modules | C: precommit asf-site-src U: . |
| Console output | https://builds.apache.org/job/yetus-github-multibranch/job/PR-62/3/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.6.0 hadolint=1.16.2-0-g3b6f5a6 markdownlint=0.15.0 |
| Powered by | Apache Yetus 0.11.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Comment by aw-was-here** [comment]
Thanks!

**Pull request merged**